src/lang/eval/EvalUtil.ml

```diff
@@ -156,16 +156,8 @@ module Configuration = struct
 
   (*  Manipulations with configuartion *)
   
-  let store st i l =
-    let k = get_id i in
-    let s = st.fields in
-    match List.find s ~f:(fun (z, _) -> z = k) with
-    | Some (_, l') -> pure @@
-        ({st with
-          fields = (k, l) :: List.filter ~f:(fun z -> fst z <> k) s}
-        , G_Store(l', l))
-    | None -> fail1 (sprintf "No field \"%s\" in contract.\n" k)
-          (ER.get_loc (get_rep i))
+  let store i l =
+    fromR @@ StateService.update ~fname:i ~keys:[] ~value:l
```

Comment:
This seems like a very cool replacement, but now I don't understand how the gas accounting happens. Previously, store returned a constructor that was used to subtract gas in the resource monad. Now, it's just this `StateService.update ~fname:i ~keys:[] ~value:l`. I cannot find how from the point of gas-accounting it returns the same thing as before.